playbooks/libvirt/openshift-cluster/tasks/configure_libvirt_network.yml

```diff
@@ -1,27 +1,10 @@
 ---
-- name: Test if libvirt network for openshift already exists
-  command: "virsh -c {{ libvirt_uri }} net-info {{ libvirt_network }}"
-  register: net_info_result
-  changed_when: False
-  failed_when: "net_info_result.rc != 0 and 'no network with matching name' not in net_info_result.stderr"
-
-- name: Create a temp directory for the template xml file
-  command: "mktemp -d /tmp/openshift-ansible-XXXXXXX"
-  register: mktemp
-  when: net_info_result.rc == 1
-
-- name: Create network xml file
-  template:
-    src: templates/network.xml
-    dest: "{{ mktemp.stdout }}/network.xml"
-  when: net_info_result.rc == 1
-
-- name: Create libvirt network for openshift
-  command: "virsh -c {{ libvirt_uri }} net-create {{ mktemp.stdout }}/network.xml"
-  when: net_info_result.rc == 1
-
-- name: Remove the temp directory
-  file:
-    path: "{{ mktemp.stdout }}"
-    state: absent
-  when: net_info_result.rc == 1
+- name: Create the libvirt network for OpenShift
+  virt_net:
+    name: '{{ libvirt_network }}'
+    state: '{{ item }}'
+    xml: "{{ lookup('template', 'network.xml') }}"
+    uri: '{{ libvirt_uri }}'
```

Comment:
When I tried with only the `state: active`, it failed:

```
$ cat playbooks/libvirt/openshift-cluster/tasks/configure_libvirt_network.yml
---
- name: Create the libvirt network for OpenShift
  virt_net:
    name: '{{ libvirt_network }}'
    state: active
    xml: "{{ lookup('template', 'network.xml') }}"
    uri: '{{ libvirt_uri }}'
```
leads to:
```
TASK [Create the libvirt network for OpenShift] ********************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "network openshift-ansible not found"}
```

In the [example section of the doc](https://docs.ansible.com/ansible/virt_net_module.html#examples), it is suggested that the network must be defined before being “activated”:
```
# Ensure that a network is active (needs to be defined and built first)
- virt_net: state=active name=br_nat
```

For the `autostart`, it depends whether we want the libvirt OpenShift cluster to be automatically restarted when the workstation is rebooted.